### PR TITLE
[skip ci] DEV: Remove nbsphinx and replace it with myst-nb.

### PR DESCRIPTION
**Pull request by shaneahmed**
- Remove nbsphinx and replace it with myst-nb.
- Replace Example gallery with examples/README.md.
- Remove repetitive jupyter notebook cells from docs.
- Hide outputs related to progress bar.
- Remove nbsphinx from requirements.
- Add myst-nb in requirements.

https://tia-toolbox.readthedocs.io/en/dev-doc-use-myst-nb/

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -100,7 +105,12 @@
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
 # This patterns also effect to html_static_path and html_extra_path
```
Comment (DavidBAEpstein) [inline docs/conf.py:107]: This patterns also effect to html_static_path and html_extra_path
should read
These patterns also affect html_static_path and html_extra_path

File: docs/installation.rst [hunk 2]
```diff
@@ -17,7 +17,7 @@ After `installing Anaconda <https://docs.anaconda.com/anaconda/install/index.htm
 
     $ conda install -c conda-forge tiatoolbox
 
-Please note that conda-forge installation support is limited on Windows as openslide binaries are not supported on official conda channels. An alternate way to install using conda on Windows could be to install it in `WSL2 with CUDA support <https://docs.microsoft.com/en-us/windows/ai/directml/gpu-cuda-in-wsl>`_.
+Please note that conda-forge installation support is limited on Windows as openslide binaries are not supported on official conda channels. An alternate way to install using conda on Windows could be to install it in `WSL2 with CUDA support <https://docs.microsoft.com/en-us/windows/ai/directml/gpu-cuda-in-wsl>`_. We have found WSL2 to be faster to run Python codes in some cases and therefore we would recommend this option.
```
Comment (DavidBAEpstein) [inline docs/installation.rst:20]: We have found WSL2 to be faster to run Python codes in some cases and therefore we would recommend this option.

This is OK, but I prefer:
In some cases, WSL2 runs faster on Python codes iand therefore we recommend this option.

**Review by DavidBAEpstein — APPROVED** [review]
This PR must have involved a great deal of work. A professional job well done.
I have made a few comments which I hope will be accepted, but I approve the PR in any case.

File: docs/conf.py (hunk 1 above)
Comment (shaneahmed) [reply docs/conf.py]: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/454/commits/1ce506606220a371d145c7e4e0b3f1a271b71103

File: docs/installation.rst (hunk 2 above)
Comment (shaneahmed) [reply docs/installation.rst]: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/454/commits/0fe39b442fe592ee78767aec2c26e62dc8cac51b

**Comment by shaneahmed** [comment]
@John-P This PR is now ready for final checks.

File: docs/installation.rst [hunk 3]
```diff
@@ -17,7 +17,7 @@ After `installing Anaconda <https://docs.anaconda.com/anaconda/install/index.htm
 
     $ conda install -c conda-forge tiatoolbox
 
-Please note that conda-forge installation support is limited on Windows as openslide binaries are not supported on official conda channels. An alternate way to install using conda on Windows could be to install it in `WSL2 with CUDA support <https://docs.microsoft.com/en-us/windows/ai/directml/gpu-cuda-in-wsl>`_.
+Please note that conda-forge installation support is limited on Windows as openslide binaries are not supported on official conda channels. An alternate way to install using conda on Windows could be to install it in `WSL2 with CUDA support <https://docs.microsoft.com/en-us/windows/ai/directml/gpu-cuda-in-wsl>`_. In some cases, WSL2 runs faster on Python codes and therefore we recommend this option.
```
Comment (John-P) [inline docs/installation.rst:20]: Might want to bold the WSL2 bit as it is easy to miss.

**Review by John-P — APPROVED** [review]
These changes look great and work really well with the new theme.

File: docs/installation.rst (hunk 3 above)
Comment (shaneahmed) [reply docs/installation.rst]: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/454/commits/4bfe73945e096dc0b7d07e8d211faeda1289c9a9

**Pull request merged**